core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java

```diff
@@ -527,7 +527,7 @@ protected DataBuffer getDataBuffer(BlockReadRequestContext context, long offset,
           }
         } else {
           ByteBuffer buf = blockReader.read(offset, len);
-          return new NettyDataBuffer(Unpooled.wrappedBuffer(buf));
+          return new PooledNioDataBuffer(buf, len);
```

Comment:
good call!   I prefer your second idea -- changing the blockReader interface to accept pre-allocated byte buffer or accept a supplier of the bytebuffer.   But I will make in a separate PR.   Since the interface change is on a critical path, I don't want to mix it with this pr.

For this pr, I will just make use of the new PooledNioBuffer for paged block store and keep the old code for the mono block store.   Is that ok?

Reply:
LGTM. Maybe add a todo?